lib/workers/repository/process/lookup/index.ts

```diff
@@ -84,14 +83,17 @@ export async function lookupUpdates(
         return res;
       }
 
-      const { val: lookupValue, err: lookupError } = await Result.wrapNullable(
-        getPkgReleases(config),
-        'no-releases' as const
+      const { val: releaseResult, err: lookupError } = await getRawPkgReleases(
+        config
       )
-        .transform((x) => Result.ok(clone(x)))
+        .transform((res) => applyDatasourceFilters(res, config))
```

Comment:
why should we do filtering here instead of datasource?

Reply:
Because I want to propagate all the filtering 1-2 layers up, in order to implement:
- #6575

Reply:
🤔 I heavily use this in internal-tool, how can we make it public available in future? it can of cause be breaking in kind of API change, but I need the filtering via allowed and extract version.

Reply:
You still can use `getPkgReleases`, I left it unchanged for using outside of the repo. Or what do you mean?